alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt

```diff
@@ -109,6 +115,13 @@ class TransportGetAlertsAction @Inject constructor(
         } else if (getAlertsRequest.monitorIds.isNullOrEmpty() == false) {
             queryBuilder.filter(QueryBuilders.termsQuery("monitor_id", getAlertsRequest.monitorIds))
         }
+        if (getAlertsRequest.workflowIds.isNullOrEmpty() == false) {
+            val bqb: BoolQueryBuilder = QueryBuilders.boolQuery()
+            queryBuilder.must(QueryBuilders.termsQuery("workflow_id", getAlertsRequest.workflowIds))
+            if (getAlertsRequest.monitorId.isNullOrEmpty() && getAlertsRequest.monitorIds.isNullOrEmpty()) {
+                queryBuilder.must(QueryBuilders.termQuery("monitor_id", ""))
```

Comment:
Can this be applied even when `workflowIds` is empty? If not, then wouldnt that mean this would try to get workflow alerts?